unlock-app/src/components/interface/checkout/Stepper.tsx

```diff
@@ -79,22 +81,26 @@ export interface StepItem {
 }
 
 interface StepperProps {
-  items: StepItem[]
-  position: number
   service: CheckoutService | UnlockAccountService
   disabled?: boolean
 }
 
-export const Stepper = ({
-  items,
-  position,
-  service,
-  disabled,
-}: StepperProps) => {
-  const index = items.findIndex((item) => item.id === position)
+export const Stepper = ({ service, disabled }: StepperProps) => {
+  const [state] = useActor(service)
+
+  const isUnlockAccount = service.id === 'unlockAccount'
+  // @ts-expect-error Property 'renew' does not exist on type 'UnlockAccountMachineContext'.
+  const isRenew = service.id === 'checkout' && !!state.context?.renew
+
+  const items = useStepperItems(service, { isUnlockAccount, isRenew })
+
+  const index = items.findIndex(
```

Comment:
Why not use find instead of find index?

Reply:
becuase we actually need the index to construct the `base` and `rest` arrays...